### ISSUE TITLE
Cannot see videos in dashboard

### ISSUE THREAD

**Issue by SilverXXX** [zalando/zalenium#847]
## 🐛 Bug Report

When accessing the dashboard, a list of video is shown but there is no playback, and many logs with stacktraces are written

## To Reproduce

Install with :
helm install --set hub.maxDockerSeleniumContainers=100 --set hub.desiredContainers=10 --set ingress.enabled=true --name zalenium-name charts\zalenium\
then launching any test

## Expected behavior

Videos should be playing inside the dashboard

## Actual behavior

Videos are not showing at all, and these lines appears inside pod logs:
13:29:38.402 [http://10.40.0.40:40000] WARN  d.z.e.z.p.DockerSeleniumRemoteProxy - Error while copying the video
java.nio.file.NoSuchFileException: /home/seluser/videos/build_2019_01_22T13_28_15_280Z/zalenium_webRtcOperator_1_firefox_LINUX_20190122132832642_FAILED.mp4
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at java.nio.file.Files.copy(Files.java:3016)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.copyVideos(DockerSeleniumRemoteProxy.java:639)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.processContainerAction(DockerSeleniumRemoteProxy.java:609)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.processContainerAction(DockerSeleniumRemoteProxy.java:598)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.videoRecording(DockerSeleniumRemoteProxy.java:562)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.cleanupNode(DockerSeleniumRemoteProxy.java:749)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.afterSession(DockerSeleniumRemoteProxy.java:397)
	at org.openqa.grid.internal.TestSlot.performAfterSessionEvent(TestSlot.java:253)
	at de.zalando.ep.zalenium.registry.ZaleniumRegistry._release(ZaleniumRegistry.java:198)
	at de.zalando.ep.zalenium.registry.ZaleniumRegistry.lambda$0(ZaleniumRegistry.java:182)
	at java.lang.Thread.run(Thread.java:748)
13:29:39.401 [OkHttp https://10.96.0.1/...] ERROR i.f.k.c.d.i.ExecWebSocketListener - Exec Failure: HTTP:0. Message:No response
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.kubernetes.client.dsl.internal.ExecWebSocketListener.onMessage(ExecWebSocketListener.java:266)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:326)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:221)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:213)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Read end dead
	at java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:262)
	at java.io.PipedInputStream.awaitSpace(PipedInputStream.java:268)
	at java.io.PipedInputStream.receive(PipedInputStream.java:231)
	at java.io.PipedOutputStream.write(PipedOutputStream.java:149)
	at java.io.OutputStream.write(OutputStream.java:75)
	at io.fabric8.kubernetes.client.dsl.internal.ExecWebSocketListener.onMessage(ExecWebSocketListener.java:248)
	... 10 common frames omitted
13:29:39.401 [OkHttp https://10.96.0.1/...] ERROR d.z.e.z.c.k.KubernetesContainerClient - zalenium-40000-hlqgc Failed to execute command [tar, -C, /videos/, -c, .]
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.kubernetes.client.dsl.internal.ExecWebSocketListener.onMessage(ExecWebSocketListener.java:266)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:326)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:221)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:273)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:213)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Read end dead
	at java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:262)
	at java.io.PipedInputStream.awaitSpace(PipedInputStream.java:268)
	at java.io.PipedInputStream.receive(PipedInputStream.java:231)
	at java.io.PipedOutputStream.write(PipedOutputStream.java:149)
	at java.io.OutputStream.write(OutputStream.java:75)
	at io.fabric8.kubernetes.client.dsl.internal.ExecWebSocketListener.onMessage(ExecWebSocketListener.java:248)
	... 10 common frames omitted

## Environment

OS: ContainerOs
Zalenium Image Version(s): zalenium-3.141.59g
Docker version: Docker version 17.03.2-ce, build 2360430
Kubernetes:
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:08:19Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

**Comment by diemol** [comment]
@SilverXXX 
Could you please provide a self contained test without any dependencies to reproduce the issue? A GitHub repo link would be ideal.

**Comment by diemol** [comment]
I have not tried it yet, but I wanted to ask you if it works if you omit the "build" capability? Or if you set the "build" capability to a hardcoded value.

**Comment by SilverXXX** [comment]
Leaving "build" completely out make things work again, but using a fixed value (e.g. "test1") breaks them.

**Comment by marcos-alfredo** [comment]
I have a similar issue where the container is never removed and no video is shown if the build name is set.
Is that a known issue?

**Comment by BogdanLivadariu** [comment]
I can confirm the issue happens for me as well.
passing `build` capability results in the zalenium host when uploading videos
```
05:59:00.403 [http://10.0.0.153:40000] WARN  d.z.e.z.p.DockerSeleniumRemoteProxy - Error while copying the video
java.nio.file.NoSuchFileException: /home/seluser/videos/jenkins_zzzz/49f2a383_da33_4a6d_8e11_269eab66d196_chrome_LINUX_SUCCESS.mp4
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at java.nio.file.Files.copy(Files.java:3016)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.copyVideos(DockerSeleniumRemoteProxy.java:639)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.processContainerAction(DockerSeleniumRemoteProxy.java:609)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.processContainerAction(DockerSeleniumRemoteProxy.java:598)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.videoRecording(DockerSeleniumRemoteProxy.java:562)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.cleanupNode(DockerSeleniumRemoteProxy.java:749)
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.afterSession(DockerSeleniumRemoteProxy.java:397)
	at org.openqa.grid.internal.TestSlot.performAfterSessionEvent(TestSlot.java:253)
	at de.zalando.ep.zalenium.registry.ZaleniumRegistry._release(ZaleniumRegistry.java:198)
	at de.zalando.ep.zalenium.registry.ZaleniumRegistry.lambda$0(ZaleniumRegistry.java:182)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by pearj** [comment]
I tried running the test with `THREAD_NUM=1 GRID_URL=http://zalenium-zalenium.192.168.99.100.nip.io/wd/hub ./gradlew clean test`
, and it fails on my local OpenShift. So it does look like a bug.

**Comment by pearj** [comment]
I found the bug, @diemol tidied some things up, but didn't realise that when using a `buildName` the `testInformation.getFileName()` actually contains a directory, which is probably a bug in itself.

It was introduced here: https://github.com/zalando/zalenium/commit/9975aea4ef3befba2f6c035d6feb377b752ad076#diff-cdb17f721a30ede405aeccaa422c3006R634

**Comment by diemol** [comment]
I'm reworking the dashboard now and I'll have a look at this tomorrow. Thanks for checking.

**Comment by diemol** [comment]
Fixed via #851, will be released in the next couple of days.
